app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,14 @@
+class SessionsController < ApplicationController
+  def create
+    if request.env['omniauth.auth']
+      user = User.create_with_omniauth(request.env['omniauth.auth'])
+      session[:user_id] = user.id
+      redirect_to user_path(user.id)
+    else
+      user = User.find_by_email(params[:email])
+      user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect_to user_path(user.id)
```

Comment:
can put these last two lines outside of the if/else since they happen in both branches.

Reply:
does `SecurePassword#authenticate` raise an error that'll respond with the appropriate http status?